File: system-test/java/src/main/java/com/optivem/shop/dsl/port/ScenarioDsl.java
package com.optivem.shop.dsl.port;

import com.optivem.shop.dsl.port.assume.AssumeStage;
import com.optivem.shop.dsl.port.given.GivenStage;
import com.optivem.shop.dsl.port.when.WhenStage;

public interface ScenarioDsl {
    AssumeStage assume();

    GivenStage given();

    WhenStage when();
}

File: system-test/typescript/src/dsl/scenario/scenario-dsl.ts
import { AppContext } from './app-context';
import { ScenarioContext } from './scenario-context';
import { AssumeStage } from './assume/assume-stage';
import { GivenStage } from './given/given-stage';
import { WhenStage } from './when/when-stage';

export class ScenarioDsl {
  constructor(private readonly app: AppContext) {}

  assume(): AssumeStage {
    return new AssumeStage(this.app);
  }

  given(): GivenStage {
    return new GivenStage(this.app, new ScenarioContext());
  }

  when(): WhenStage {
    return new WhenStage(this.app, new ScenarioContext());
  }

  async close(): Promise<void> {
    await this.app.closeAll();
  }
}
